pkg/endpoint/policy.go

```diff
@@ -408,59 +409,61 @@ func (e *Endpoint) updateRegenerationStatistics(context *regenerationContext, er
 func (e *Endpoint) Regenerate(owner Owner, regenMetadata *ExternalRegenerationMetadata) <-chan bool {
 	done := make(chan bool, 1)
 
-	go func() {
-		var buildSuccess bool
+	regenContext := regenMetadata.toRegenerationContext()
 
-		regenContext := regenMetadata.toRegenerationContext()
+	epEvent := eventqueue.NewEvent(&EndpointRegenerationEvent{
+		owner:        owner,
+		regenContext: regenContext,
+		ep:           e,
+	})
 
-		defer func() {
-			done <- buildSuccess
-			close(done)
-		}()
+	// This may block if the Endpoint's EventQueue is full.
+	resChan := e.EventQueue.Enqueue(epEvent)
```

Comment:
AFAICT we previously put all of the work in this function into a parallel goroutine. If this blocks, will it cause any problems to the callers? (Do we have a good way to stress-test this?)

Reply:
(Oh, this is called from policy package right? so this is deliberately put outside the goroutine to ensure it's run from the policy event handler goroutine? If so, might be good to note this here as well so someone doesn't go and subtly change the event handling model with a seemingly innocuous change)

Reply:
Added comment to indicate that callers depend on this being synchronous.